proposals/2312-matrix-uri.md

```diff
@@ -356,6 +356,95 @@ Client authors are strongly encouraged to standardise custom query elements
 that gain adoption by submitting an MSC defining them in a way compatible
 across the client ecosystem.
 
+We define four other query parameters below which improve the ability to onboard
+new matrix users (`default_client` and `default_hs`) and let link authors
+specify a mandated client or server (`force_client` and `force_hs`).
```

Comment:
I don't think these would be used that extensively.  The use cases are respectively for "branded links" from web pages for existing communities (rather than on billboards or napkins), or for more exotic "this link requires you to be using a specific client or server" (which would be very bizarre to put on billboards or napkins) use cases.

Reply:
An interesting case for `default_` emerged in the Matrix URI room: there's an owner of a DIY- (family-, community-) homeserver installation and he/she wants to onboard some people to it because they ask him/her about an alternative server ("ok, I get it that using matrix.org is not exactly kosher in a decentralised network, which one would you suggest instead?"). What kind of link should he/she send to the invitees to specifically onboard them on a new homeserver and nothing else? There's no particular room to invite to; those people are assumed to already be more or less at home in Matrix. Seems like allowing an empty path for a case like `matrix:?default_hs=mylittlehs.net` would make sense in that case?